### PR TITLE
chore(Geo): add error handling to device tracking

### PR DESCRIPTION
**Pull request by phantumcode**
*Issue #, if available:*

*Description of changes:*
* Add error handling to device tracking and send error to Amplify Hub as an event
*Check points: (check or cross out if not relevant)*

- [ ] ~Added new tests to cover change, if needed~
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

*DataStore checkpoints (check when completed)*

- [ ] Ran AWSDataStorePluginIntegrationTests
- [ ] Ran AWSDataStorePluginV2Tests
- [ ] Ran AWSDataStorePluginMultiAuthTests
- [ ] Ran AWSDataStorePluginCPKTests
- [ ] Ran AWSDataStorePluginAuthCognitoTests
- [ ] Ran AWSDataStorePluginAuthIAMTests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift [hunk 1]
```diff
@@ -177,14 +179,19 @@ class AWSDeviceTracker: NSObject, CLLocationManagerDelegate, AWSDeviceTrackingBe
...
+                error)
+            sendHubErrorEvent(error: geoError, locations: receivedLocations)
         }
     }
 
     func mapReceivedLocationsToPositions(receivedLocations: [CLLocation], currentTime: Date) -> [Position] {
         guard let deviceId = UserDefaults.standard.object(forKey: AWSDeviceTracker.deviceIDKey) as? String else {
             Amplify.log.error("Not able to fetch deviceId from UserDefaults")
-            // TODO: send error on hub
+            let error = Geo.Error.unknown(
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift:192]: should we add a new error constant for this scenario?

File: Amplify/Categories/Geo/GeoCategory+HubPayloadEventName.swift [hunk 2]
```diff
@@ -9,3 +9,7 @@ public extension HubPayload.EventName {
     /// Geo hub events
     struct Geo { }
 }
+
+public extension HubPayload.EventName.Geo {
+    static let saveLocationFailed = "Geo.saveLocationFailed"
```
[the remark this answers is not in the document]
Comment (phantumcode) [reply Amplify/Categories/Geo/GeoCategory+HubPayloadEventName.swift]: Updated per suggestion

File: AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Support/Constants/GeoPluginErrorConstants.swift [hunk 3]
```diff
@@ -32,6 +32,14 @@ struct GeoPluginErrorConstants {
     static let errorInitializingLocalStore: GeoPluginErrorString = (
         "Unable to create local database for storing locations.",
         "This seems like a bug. Please report an issue on Amplify Swift repo.")
+    
+    static let errorSavingLocationsToLocalStore: GeoPluginErrorString = (
+        "Unable to save locations to local database.",
+        "This seems like a bug. Please report an issue on Amplify Swift repo.")
+    
+    static let errorSaveLocationsFailed: GeoPluginErrorString = (
+        "Unable to save locations to service.",
```
[the remark this answers is not in the document]
Comment (phantumcode) [reply AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Support/Constants/GeoPluginErrorConstants.swift]: Updated per suggestion

File: AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift [hunk 4]
```diff
@@ -229,13 +240,21 @@ class AWSDeviceTracker: NSObject, CLLocationManagerDelegate, AWSDeviceTrackingBe
...
+                         Amplify.log.error("Unable to convert stored locations to positions.")
+                         let geoError = Geo.Error.unknown(
+                             GeoPluginErrorConstants.errorSaveLocationsFailed.errorDescription, GeoPluginErrorConstants.errorSaveLocationsFailed.recoverySuggestion,
+                             error)
+                         sendHubErrorEvent(error: geoError, locations: positions)
+                        throw GeoErrorHelper.mapAWSLocationError(error)
+                     }
                 } catch {
-                    // TODO: send error on hub
+                    Amplify.log.error("Unable to convert stored locations to positions.")
```
[the remark this answers is not in the document]
Comment (phantumcode) [reply AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift]: Log message updated.

File: AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift (hunk 1 above)
Comment (phantumcode) [reply AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift]: I added a new `Geo.Error.internalPluginError`

**Review by thisisabhash — APPROVED** [review]

**Pull request merged**